pkg/scheduler/schedule_one.go

```diff
@@ -946,7 +946,11 @@ func updatePod(ctx context.Context, client clientset.Interface, pod *v1.Pod, con
 	// NominatedNodeName is updated only if we are trying to set it, and the value is
 	// different from the existing one.
 	nnnNeedsUpdate := nominatingInfo.Mode() == framework.ModeOverride && pod.Status.NominatedNodeName != nominatingInfo.NominatedNodeName
-	if !podutil.UpdatePodCondition(podStatusCopy, condition) && !nnnNeedsUpdate {
+	condNeedsUpdate := podutil.UpdatePodCondition(podStatusCopy, condition)
+	totalResourcesNeedsUpdate := podutil.UpdateTotalResources(podStatusCopy, pod)
+
+	// we don't need to update condition, nominated node name or pod total resources
+	if !condNeedsUpdate && !nnnNeedsUpdate && !totalResourcesNeedsUpdate {
 		return nil
 	}
 	if nnnNeedsUpdate {
```

Comment:
nit: maybe this can follow the pattern and actually make an update before the previous `if` condition.